### PR TITLE
doc: rpc-mining: Clarify error messages

### LINKED ISSUE #13274: getblocktemplate fails in regtest mode even when it is connected to the other node
I have created two nodes and connected the 1st node with the 2nd node in regtest mode . Now when I make the getblocktemplate command using bitcoin-cli . It fails with bitcoin is downloading the block.
 
Bitcoin core vs v0.16

_admin1@root:~/Projects/bitcoin$ ./src/bitcoin-cli -regtest -port=8333 -rpcport=8332 -rpcuser=test -rpcpassword=test getconnectioncount_
**1**
_admin1@root:~/Projects/bitcoin$ ./src/bitcoin-cli -regtest -port=8333 -rpcport=8332 -rpcuser=test -rpcpassword=test getblocktemplate_
**error code: -10
error message:
Bitcoin is downloading blocks.**

### PR DESCRIPTION
**Pull request by maflcko**
Fixes #13274

### CODE REVIEW

**Comment by practicalswift** [comment]
ACK fa642d770e4606e005fc41e8155ed48d2a9c5532

File: src/rpc/mining.cpp [hunk 1]
```diff
@@ -444,10 +444,10 @@ static UniValue getblocktemplate(const JSONRPCRequest& request)
         throw JSONRPCError(RPC_CLIENT_P2P_DISABLED, "Error: Peer-to-peer functionality missing or disabled");
 
     if (g_connman->GetNodeCount(CConnman::CONNECTIONS_ALL) == 0)
-        throw JSONRPCError(RPC_CLIENT_NOT_CONNECTED, "Bitcoin is not connected!");
+        throw JSONRPCError(RPC_CLIENT_NOT_CONNECTED, "Bitcoin Core is not connected!");
```
Comment (luke-jr) [inline src/rpc/mining.cpp]: Should be using `PACKAGE_NAME`

**Comment by MarcoFalke** [comment]
Using PACKAGE_NAME:

Screenshot from 2019-03-28 09-26-48

File: src/rpc/mining.cpp [hunk 2]
```diff
@@ -444,10 +444,10 @@ static UniValue getblocktemplate(const JSONRPCRequest& request)
         throw JSONRPCError(RPC_CLIENT_P2P_DISABLED, "Error: Peer-to-peer functionality missing or disabled");
 
     if (g_connman->GetNodeCount(CConnman::CONNECTIONS_ALL) == 0)
-        throw JSONRPCError(RPC_CLIENT_NOT_CONNECTED, "Bitcoin is not connected!");
+        throw JSONRPCError(RPC_CLIENT_NOT_CONNECTED, std::string{PACKAGE_NAME} + " is not connected!");
```
Comment (luke-jr) [inline src/rpc/mining.cpp]: Pretty sure you could do just:

```C++
        throw JSONRPCError(RPC_CLIENT_NOT_CONNECTED, PACKAGE_NAME " is not connected!");
```

but this works too.
￼

Comment (MarcoFalke) [reply src/rpc/mining.cpp]: Used this for brevity

**Review by luke-jr — APPROVED** [review]

**Comment by fanquake** [comment]
tACK fa292ad

Screen Shot 2019-04-03 at 10 42 42 pm

**Pull request merged**